### PR TITLE
Fixes #5741: Backport the BDB bundling in slapd from master

### PR DESCRIPTION
**Pull request by Kegeruneku**
Ticket: http://www.rudder-project.org/redmine/issues/5741

To be merged in 2.10

### CODE REVIEW

**Comment by Kegeruneku** [comment]
bump

**Comment by jooooooon** [comment]
@peckpeck Could you review and merge this if OK please?

For context: we decided to bundle our own BerkeleyDB libs to avoid depending on those from the system (which causes an almost impossible upgrade situation when you upgrade the OS which upgrades both BerkeleyDB and Rudder simultaneously). This was done in the 3.0 branch, and this PR is to backport the change to the 2.10 branch to avoid this problem in stable versions.

File: rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec [hunk 1]
```diff
@@ -103,22 +104,40 @@ rudder-agent package installed) and for configuration rules and parameters.
...
-cd openldap-source
 
 # Ensure an appropriate environment for the compiler
 export CFLAGS="$RPM_OPT_FLAGS"
 export CXXFLAGS="$RPM_OPT_FLAGS"
 
-./configure --build=%_target --prefix=%{rudderdir} --enable-dynamic --enable-debug --enable-modules --enable-hdb=mod --enable-monitor=mod --enable-dynlist=mod --with-cyrus-sasl
+# 1 - BerkeleyDB
+cd berkeleydb-source/build_unix/
+../dist/configure --build=%_target --prefix=%{rudderdir}
```
Comment (peckpeck) [inline rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec]: %{_target}

File: rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec [hunk 2]
```diff
@@ -103,22 +104,40 @@ rudder-agent package installed) and for configuration rules and parameters.
...
+
+# 2 - OpenLDAP
+cd openldap-source
+
+export LD_LIBRARY_PATH="/opt/rudder/lib"
+export CPPFLAGS="-I/opt/rudder/include"
+export LDFLAGS="-L/opt/rudder/lib"
+export LIBS="-L/opt/rudder/lib"
+
+./configure --build=%_target --prefix=%{rudderdir} --libdir=%{rudderdir}/lib/ldap --enable-dynamic --enable-debug --enable-modules --enable-hdb=mod --enable-monitor=mod --enable-dynlist=mod --with-cyrus-sasl
```
Comment (peckpeck) [inline rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec]: what is this "--libdir=%{rudderdir}/lib/ldap " for ?

File: rudder-inventory-ldap/debian/rules [hunk 3]
```diff
@@ -9,23 +9,21 @@
...
-
-
-
 configure: configure-stamp
 configure-stamp:
 	dh_testdir
 	# Add here commands to configure the package.
-	cd SOURCES/openldap-source && ./configure --prefix=/opt/rudder --enable-dynamic --enable-debug --enable-modules --enable-hdb=mod --enable-monitor=mod --enable-dynlist=mod --with-cyrus-sasl
+	cd SOURCES/berkeleydb-source/build_unix && ../dist/configure --prefix=/opt/rudder
+	export LD_LIBRARY_PATH="/opt/rudder/lib" ; CPPFLAGS="-I/opt/rudder/include" ; export LDFLAGS="-L/opt/rudder/lib" ; export LIBS="-L/opt/rudder/lib" ; cd SOURCES/openldap-source && ./configure --prefix=/opt/rudder --libdir=/opt/rudder/lib/ldap --enable-dynamic --enable-debug --enable-modules --enable-hdb=mod --enable-monitor=mod --enable-dynlist=mod --with-cyrus-sasl
```
Comment (peckpeck) [inline rudder-inventory-ldap/debian/rules]: A more posix compliant version would be :

cd SOURCES/openldap-source && VAR1=bla VAR2=bla ./configure ...

**Comment by jooooooon** [comment]
@Kegeruneku Ping! Please address these remarks.

**Comment by Kegeruneku** [comment]
Ja !

File: rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec (hunk 2 above)
Comment (Kegeruneku) [inline rudder-inventory-ldap/SPECS/rudder-inventory-ldap.spec]: To store all the OpenLDAP related libs in /opt/rudder/lib/ldap to make things cleaner :)

**Comment by peckpeck** [comment]
Good !

**Pull request merged**